jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/util/InputStreamResponseListener.java

```diff
@@ -256,7 +255,16 @@ public InputStream getInputStream()
         InputStream result = new Input();
         if (stream.compareAndSet(null, result))
             return result;
-        return IO.getClosedStream();
+        result = InputStream.nullInputStream();
+        try
+        {
+            result.close();
+        }
+        catch (IOException ignore)
+        {
+            // not possible with InputStream.nullInputStream()
+        }
```

Comment:
Why this `try/catch` block?

Reply:
Because `InputStream.close()` throws `IOException`, and your `InputStreamResponseListener.getInputStream()` does not.

* https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/InputStream.html#close()

Reply:
The Javadoc for `InputStreamListener.getInputStream()` says ..

> * The method may be invoked only once; subsequent invocations will return a closed {@link InputStream}.

This makes sure that the API contract is still satisfied, _and_ also ensures that use of this "second use" InputStream returned correctly produces IOExceptions on any of the read methods.

Reply:
@sbordet bump, are you ok with this now? (the code changed here too, a better approach proposed by @gregw)

Reply:
```
$ jshell
|  Welcome to JShell -- Version 11.0.16
|  For an introduction type: /help intro

jshell> InputStream.nullInputStream();
$1 ==> java.io.InputStream$1@3d646c37

jshell> $1.read()
$2 ==> -1

jshell> $1.close()

jshell> $1.read()
|  Exception java.io.IOException: Stream closed
|        at InputStream$1.ensureOpen (InputStream.java:86)
|        at InputStream$1.read (InputStream.java:98)
|        at (#4:1)

jshell> /exit
|  Goodbye
```

Reply:
```
$ jshell
|  Welcome to JShell -- Version 11.0.16
|  For an introduction type: /help intro

jshell> new BufferedInputStream(new ByteArrayInputStream(new byte[] { 0x01 }))
$1 ==> java.io.BufferedInputStream@26a1ab54

jshell> $1.read()
$2 ==> 1

jshell> $1.close()

jshell> $1.read()
|  Exception java.io.IOException: Stream closed
|        at BufferedInputStream.getBufIfOpen (BufferedInputStream.java:176)
|        at BufferedInputStream.fill (BufferedInputStream.java:220)
|        at BufferedInputStream.read (BufferedInputStream.java:271)
|        at (#4:1)

jshell> /exit
|  Goodbye
```

Reply:
and ...

```
$ jshell
|  Welcome to JShell -- Version 11.0.16
|  For an introduction type: /help intro

jshell> var fi = new FileInputStream(new File("/tmp/foo.txt"))
fi ==> java.io.FileInputStream@26a1ab54

jshell> fi.read()
$2 ==> 70

jshell> fi.close()

jshell> fi.read()
|  Exception java.io.IOException: Stream Closed
|        at FileInputStream.read0 (Native Method)
|        at FileInputStream.read (FileInputStream.java:231)
|        at (#4:1)

jshell> /exit
|  Goodbye
```